src/main/java/hudson/plugins/claim/AbstractAssignedClaimsReport.java

```diff
@@ -0,0 +1,127 @@
+import hudson.model.Run;
+import hudson.model.View;
+import hudson.tasks.test.AbstractTestResultAction;
+import hudson.tasks.test.TestResult;
+import hudson.util.RunList;
+import jenkins.model.Jenkins;
+
+public class AbstractAssignedClaimsReport implements Action, IconSpec {
+    @Override
+    public String getIconClassName() {
+        return "icon-claim-claim";
+    }
+
+    public String getIconFileName() {
+        String iconClassName = getIconClassName();
+        if (iconClassName != null) {
+            Icon icon = IconSet.icons.getIconByClassSpec(iconClassName + " icon-md");
+            if (icon != null) {
+                JellyContext ctx = new JellyContext();
+                ctx.setVariable("resURL", Stapler.getCurrentRequest().getContextPath() + Jenkins.RESOURCE_PATH);
+                return icon.getQualifiedUrl(ctx);
+            }
+        }
+        return null;
+    }
+
+    public String getUrlName() {
+        return "/claims";
+    }
+
+    @Restricted(DoNotUse.class) // jelly only
+    public Run getFirstFail(final Run r) {
+        Run lastGood = r.getPreviousNotFailedBuild();
+        Run firstFail;
+        if (lastGood == null) {
+            firstFail = r.getParent().getFirstBuild();
+        } else {
+            firstFail = lastGood.getNextBuild();
+        }
+        return firstFail;
+    }
+
+    @Restricted(DoNotUse.class) // jelly only
+    public CommonMessagesProvider getMessageProvider(final Run r) {
+        return CommonMessagesProvider.build(getAction(r));
+    }
+
+    @Restricted(DoNotUse.class) // jelly only
+    public CommonMessagesProvider getMessageProvider(final TestResult tr) {
+        return CommonMessagesProvider.build(tr.getTestAction(ClaimTestAction.class));
+    }
+
+    public ModelObject getOwner() {
+        View view = Stapler.getCurrentRequest().findAncestorObject(View.class);
+        if (view != null) {
+            return view;
+        } else {
+            return Jenkins.getInstance().getStaplerFallback();
+        }
+    }
+
+    private ClaimBuildAction getAction(final Run r) {
+        return ClaimUtils.getBuildAction(r, false);
+    }
+
+    public RunList getBuilds() {
+        List<Run> lastBuilds = new ArrayList<>();
+        for (Job job : getJobs()) {
+            Run lb = job.getLastCompletedBuild();
+            if (lb != null && isDisplayed(lb.getAction(ClaimBuildAction.class))) {
+                lastBuilds.add(lb);
+            }
+        }
+
+        return RunList.fromRuns(lastBuilds).failureOnly();
+    }
+
+	protected List<Job> getJobs() {
+		return Jenkins.getInstance().getAllItems(Job.class);
+	}
+
+    public List<TestResult> getTestFailuresWithClaim() {
+        List<TestResult> claimedTestFailures = new ArrayList<>();
+        for (Job job : getJobs()) {
+            Run lb = job.getLastCompletedBuild();
+            if (lb != null) {
+				AbstractTestResultAction testResultAction = lb.getAction(AbstractTestResultAction.class);
```

Comment:
Could you please fix the formatting, and make sure you use spaces and not tabs in files ?

Reply:
Done